### ISSUE TITLE
EEGSym deviates from the reference architecture — can't load the authors' pretrained weights (params +26%)

### ISSUE THREAD

**Issue by bruAristimunha** [braindecode/braindecode#1088]
## Summary

`braindecode.models.EEGSym` does not reproduce the reference architecture of Pérez-Velasco et al., 2022 (*EEGSym: Overcoming Inter-Subject Variability in MI-Based BCIs With Deep Learning*, IEEE TNSRE; authors' code: https://github.com/Serpeve/EEGSym). Several departures change the actual network: total parameters are **181,606** vs the reference 8-electrode **144,440** (+26%), and the authors' published `EEGSym_pretrained_weights_{8,16}_electrode.h5` — the basis of the paper's 84.5 % (8-ch) / 88.6 % (16-ch) Physionet transfer-learning results — **cannot be loaded** into the current module.

To pin this down I built a standalone faithful port directly from `EEGSym_architecture.py` + the `.h5`. It is **forward-equivalent to the authors' Keras model** (`max|Δ| = 8.9e-8` on a fixed input, with the pretrained weights loaded), which confirms the divergences below are real rather than cosmetic. On PhysionetMI cross-subject LOSO (from scratch, 17 held-out subjects) the corrected architecture also scores **+1.3 pp** (70.6 vs 69.4).

## Divergences — `models/eegsym.py` vs `EEGSym_architecture.py` / the `.h5`

| # | braindecode | reference / `.h5` | where |
|---|---|---|---|
| 1 | `scales_samples = [int(s*sfreq/2000)*2+1]` → `[65,33,17]` (forced odd, descending) | `[int(s*fs/1000)]` → `[64,32,16]`; weights trained **ascending** `[16,32,64]` | ~L204 |
| 2 | residual filter progression `72, 36, 18` ("No reduction") | `36, 36, 18` (first unit reduces to `fpb*len//2`) | ~L276–301 |
| 3 | residual-branch 1×1×1 projection added only when `in != out`, so the 36→36 `unit_conv_1` (a whole Conv+BN) is **omitted** | reference always applies it (`.h5` `conv3d_75/79/83`) | ~L604 |
| 4 | single-scale residual spatial conv is **grouped** depthwise (`groups=filters`) | reference is a **dense** conv (`.h5` `conv3d_74/78/82`, groups=1) | ~L624–633 |
| 5 | channel-merging grouped conv `groups = fpb*len//12` (=6) | `fpb*len//8` (=9; `.h5` `conv3d_87` ⇒ in//groups=2) | ~L327 |
| 6 | `spatial_resnet_repetitions` default **5** | reference default **1** (the pretrained config) | ~L179 |
| 7 | forces **odd** kernels + symmetric `same` padding | **even** kernels with Keras asymmetric `same` (`pad_left=(k-1)//2`, `pad_right=k//2`) | — |
| 8 | `BatchNorm3d` default `eps=1e-5` | Keras `eps=1e-3` (so trained running stats apply with the wrong eps) | — |

(1) and (7) are linked: forcing odd kernels appears to be a way to use PyTorch's symmetric `same` padding, but it changes the trained kernel sizes and — together with the descending order — reverses the multi-scale concatenation, mis-aligning every downstream conv even where per-conv shapes still match.

## Reproduction (parameter-count mismatch)

```python
import h5py, numpy as np, torch
from braindecode.models import EEGSym
chs = [{"ch_name": n} for n in ["F3","C3","P3","Cz","Pz","F4","C4","P4"]]
m = EEGSym(n_outputs=2, n_times=384, sfreq=128, chs_info=chs, filters_per_branch=24,
           drop_prob=0.4, spatial_resnet_repetitions=1,
           left_right_chs=[["F3","F4"],["C3","C4"],["P3","P4"]], middle_chs=["Cz","Pz"])
print("braindecode:", sum(p.numel() for p in m.parameters()))      # 181606

ks = {}
h5py.File("EEGSym_pretrained_weights_8_electrode.h5").visititems(
    lambda n, o: ks.__setitem__(n, o.shape) if hasattr(o, "shape") else None)
print("reference  :", sum(int(np.prod(s)) for s in ks.values()))   # 144440
```

## Proposed fix

Align to the reference: `scales_samples = [int(s*fs/1000) for s in scales_time]` in ascending order; residual progression `36,36,18`; always include the residual projection conv; dense (not grouped) single-scale spatial conv; channel-merging `groups = fpb*len//8`; default `spatial_resnet_repetitions=1`; replicate Keras asymmetric `same` padding for the even kernels; `BatchNorm eps=1e-3`. This would also enable an optional `EEGSym.from_pretrained(...)` loading the authors' `.h5` (Conv3D `(kd,kh,kw,in//g,out)→(out,in//g,kd,kh,kw)`, Dense transpose, BN gamma/beta/moving_*→weight/bias/running_*).

I have a verified faithful implementation + a forward-equivalence-checked Keras→PyTorch loader and am happy to open a PR.

**Comment by bruAristimunha** [comment]
Follow-up: I uploaded the converted original 8-channel Keras EEGSym checkpoint to Hugging Face, so this does not need to live as an artifact in the braindecode repository: https://huggingface.co/bruAristimunha/eegsym-original-keras-8ch-pytorch

Conversion/verification summary from the standalone faithful port:
- Keras H5 SHA256: f60ddd220cf48dd18dd9706c5389c70a244fe878610021c2ed97d40105145b2b
- H5 datasets consumed: 146/146
- PyTorch tensors filled: 146/146
- state values excluding num_batches_tracked: 144440
- finite forward probe: shape (2, 2)

Given this, a minimal future braindecode PR can be limited to architecture/loading compatibility or documentation around loading from the HF checkpoint, rather than storing weights in the repo.
